### PR TITLE
Bump `pip` version to `21.2.4`

### PR DESCRIPTION
**Pull request by kaxil**
Updates `pip` version from `21.2.2` to `21.2.4`

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Review by uranusjr — APPROVED** [review]
I thought @potiuk had a PR to upgrade to 21.2.3 but maybe I misremember.

**Comment by potiuk** [comment]
Yeah. But i think it was lost somewhere :).

**Review by potiuk — APPROVED** [review]

File: IMAGES.rst [hunk 1]
```diff
@@ -492,7 +492,7 @@ The following build arguments (``--build-arg`` in docker build command) can be u
 | ``ADDITIONAL_RUNTIME_APT_ENV``           |                                          | Additional env variables defined         |
 |                                          |                                          | when installing runtime deps             |
 +------------------------------------------+------------------------------------------+------------------------------------------+
-| ``AIRFLOW_PIP_VERSION``                  | ``21.1``                                 | PIP version used.                        |
+| ``AIRFLOW_PIP_VERSION``                  | ``21.2.4``                               | PIP version used.                        |
```
Comment (kaxil) [inline IMAGES.rst:495]:
```suggestion
| ``AIRFLOW_PIP_VERSION``                  | ``21.2.4``                               | PIP version used.                      |
```

File: docs/docker-stack/build-arg-ref.rst [hunk 2]
```diff
@@ -45,7 +45,7 @@ Those are the most common arguments that you use when you want to build a custom
 +------------------------------------------+------------------------------------------+---------------------------------------------+
 | ``AIRFLOW_USER_HOME_DIR``                | ``/home/airflow``                        | Home directory of the Airflow user.         |
 +------------------------------------------+------------------------------------------+---------------------------------------------+
-| ``AIRFLOW_PIP_VERSION``                  | ``21.1``                                 | PIP version used.                           |
+| ``AIRFLOW_PIP_VERSION``                  | ``21.2.4``                                | PIP version used.                           |
```
Comment (kaxil) [inline docs/docker-stack/build-arg-ref.rst]:
```suggestion
| ``AIRFLOW_PIP_VERSION``                  | ``21.2.4``                                | PIP version used.                         |
```

**Comment by potiuk** [comment]
Building and pushing cache images now, as it is a change quite early in the Docker build process (and with still few flaky tests it is unlikely main succeeds with pushing images)

**Comment by potiuk** [comment]
Glad that we can now 'just upgrade PIP` and it works @uranusjr 👍

**Pull request merged**